File: backend/src/main/java/ru/chsergeig/shoppy/properties/SecurityProperties.java
package ru.chsergeig.shoppy.properties;

import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;

@Data
@ConfigurationProperties(prefix = "shoppy.security")
public class SecurityProperties {

    private Jwt jwt;
    private User user;

    @Data
    public static class Jwt {
        private String authorizationHeader;
        private String secret;
        private Long expirationInSeconds;
    }

    @Data
    public static class User {
        private String login;
        private String password;
    }
}

File: backend/src/main/kotlin/ru/chsergeig/shoppy/properties/SecurityProperties.kt
package ru.chsergeig.shoppy.properties

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.boot.context.properties.ConstructorBinding

@ConstructorBinding
@ConfigurationProperties(prefix = "shoppy.security")
data class SecurityProperties(
    val jwt: Jwt,
    val user: User
) {

    data class Jwt(
        val authorizationHeader: String,
        val secret: String,
        val expirationInSeconds: Long,
        val salt: String
    )

    data class User(
        val login: String,
        val password: String
    )
}
